src/precompiles.rs

```diff
@@ -260,33 +264,37 @@ pub fn ecpairing(
     gas_limit: u64,
     consumed_gas: &mut u64,
 ) -> Result<Bytes, PrecompileError> {
-    if calldata.len() % 192 != 0 {
+    if calldata.len() % ECP_INPUT_SIZE != 0 {
         return Err(PrecompileError::InvalidCalldata);
     }
     let gas_cost = ECPAIRING_STATIC_COST + ecpairing_dynamic_cost(calldata.len() as u64);
     if gas_limit < gas_cost {
         return Err(PrecompileError::NotEnoughGas);
     }
 
-    let rounds = calldata.len() / 192;
+    let rounds = calldata.len() / ECP_INPUT_SIZE;
     let mut mul: FieldElement<Degree12ExtensionField> = QuadraticExtensionFieldElement::one();
     for idx in 0..rounds {
-        let start = idx * 192;
+        let start = idx * ECP_INPUT_SIZE;
 
         // Slice lengths are checked, so unwrap is safe
-        let g1_x = BN254FieldElement::from_bytes_be(&calldata[start..start + 32]).unwrap();
-        let g1_y = BN254FieldElement::from_bytes_be(&calldata[start + 32..start + 64]).unwrap();
+        let g1_x =
+            BN254FieldElement::from_bytes_be(&calldata[start..start + ECP_FIELD_SIZE]).unwrap();
+        let g1_y = BN254FieldElement::from_bytes_be(
+            &calldata[start + ECP_FIELD_SIZE..start + ECP_FIELD_SIZE * 2],
+        )
+        .unwrap();
 
-        // G2 point: ((x_0, x_1), (y_0, y_1))
-        // both x and y have a real and an imaginary part of 32 bytes each
         let g2_x_bytes = [
-            &calldata[start + 96..start + 128],
-            &calldata[start + 64..start + 96],
+            &calldata[start + (G1_POINT_SIZE + ECP_FIELD_SIZE)
+                ..start + (G1_POINT_SIZE + ECP_FIELD_SIZE * 2)], // calldata[start + 96..start + 128]
+            &calldata[start + G1_POINT_SIZE..start + (G1_POINT_SIZE + ECP_FIELD_SIZE)], // calldata[start + 64..start + 96]
         ]
         .concat();
         let g2_y_bytes = [
-            &calldata[start + 160..start + 192],
-            &calldata[start + 128..start + 160],
+            &calldata[start + (G2_POINT_SIZE * ECP_FIELD_SIZE)
+                ..start + (G2_POINT_SIZE + ECP_FIELD_SIZE * 2)], // calldata[start + 160..start + 192]
+            &calldata[start + G2_POINT_SIZE..start + (G2_POINT_SIZE * ECP_FIELD_SIZE)], // calldata[start + 128..start + 160]
```

Comment:
This fixes `ecpairing` tests:
```suggestion
            &calldata[start + (G2_POINT_SIZE + ECP_FIELD_SIZE)
                ..start + (G2_POINT_SIZE + ECP_FIELD_SIZE * 2)], // calldata[start + 160..start + 192]
            &calldata[start + G2_POINT_SIZE..start + (G2_POINT_SIZE + ECP_FIELD_SIZE)], // calldata[start + 128..start + 160]
```

Reply:
Done!